tests/main/snap-system-key/task.yaml

```diff
@@ -0,0 +1,37 @@
+summary: Ensure security profile re-generation works with system-key
+
+execute: |
+    restart_snapd() {
+        systemctl stop snapd.service
+        while [ "$(systemctl show -pActiveState snapd.service)" != "ActiveState=inactive" ]; do
+            systemctl show -pActiveState snapd.service
+            sleep 1
+        done
+
+        systemctl start snapd.service
+        while [ "$(systemctl show -pActiveState snapd.service)" != "ActiveState=active" ]; do
+            systemctl show -pActiveState snapd.service
+            sleep 1
+        done
+    }
+
+    echo "Ensure a valid system-key file is on-disk"
+    cat /var/lib/snapd/system-key | MATCH "build-id: [0-9a-z]+"
```

Comment:
Can you please add a test that checks that apparmor features is present (regardless of the value)